osu.Game.Tests/Visual/Online/TestSceneRankingsOverlay.cs

```diff
@@ -1,38 +1,34 @@
 ﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
-using osu.Framework.Allocation;
-using osu.Game.Overlays;
 using NUnit.Framework;
-using osu.Game.Users;
 using osu.Framework.Bindables;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Overlays;
 using osu.Game.Overlays.Rankings;
+using osu.Game.Rulesets.Catch;
+using osu.Game.Users;
 
 namespace osu.Game.Tests.Visual.Online
 {
     public class TestSceneRankingsOverlay : OsuTestScene
     {
         protected override bool UseOnlineAPI => true;
 
-        [Cached(typeof(RankingsOverlay))]
-        private readonly RankingsOverlay rankingsOverlay;
+        private TestRankingsOverlay rankingsOverlay;
 
         private readonly Bindable<Country> countryBindable = new Bindable<Country>();
         private readonly Bindable<RankingsScope> scope = new Bindable<RankingsScope>();
 
-        public TestSceneRankingsOverlay()
-        {
-            Add(rankingsOverlay = new TestRankingsOverlay
-            {
-                Country = { BindTarget = countryBindable },
-                Header = { Current = { BindTarget = scope } },
-            });
-        }
+        [SetUp]
+        public void SetUp() => Schedule(loadRankingsOverlay);
 
         [Test]
-        public void TestShow()
+        public void TestParentRulesetCopiedOnInitialShow()
         {
-            AddStep("Show", rankingsOverlay.Show);
+            AddStep("set ruleset", () => Ruleset.Value = new CatchRuleset().RulesetInfo);
+            AddStep("reload rankings overlay", loadRankingsOverlay);
+            AddAssert("catch ruleset selected", () => Ruleset.Value.Equals(new CatchRuleset().RulesetInfo));
```

Comment:
This is only testing behaviour that was already working previously, and not the actual changes to it applied in this pull (namely that after the initial transfer the two ruleset bindables are entirely decoupled and not transferred on a second show again).

In other words, this test still passes after reverting the functional changes in this PR.

Reply:
Good find, the PR went through number of rewrites it became indiscernible what it's intended effects were. I'll add further coverage.

Reply:
I've added further test coverage now (also fixed fatal oversight in current code, `Ruleset.Value` being checked as if it's the rankings overlay's ruleset bindable)